### ISSUE TITLE
StructuredFilterPruner can't prune mobilenetv2

### ISSUE THREAD

**Issue by dinghuanghao** [microsoft/nni#3514]
Most convolutions cannot generate masks when use structured filter pruner to prune mobilenetv2.

I think it is because the dependency_aware=True is not well support separable-convolution.

When dependency_aware is True, a depth-wise convolution will change the 1x1 convolution's group before it, this cause the variable "pruned_per_group" to be zero.

===========My Code===========

import torch
from torchvision.models.mobilenet import mobilenet_v2
from nni.algorithms.compression.pytorch.pruning import L2FilterPruner

dummy_input = torch.ones((1, 3, 224, 224))

model = mobilenet_v2()
optimizer = torch.optim.SGD(model.parameters(), lr=0.1)

config_list = [{'sparsity': 0.5, 'op_types': ['Conv2d']}]
pruner = L2FilterPruner(model, config_list, optimizer, dependency_aware=True, dummy_input=dummy_input)
pruner.compress()

===========My Log============
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.1.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.1.conv.1
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 1,2,3,5,6,8,10,11 channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.2.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.2.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.2.conv.2,features.3.conv.2
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 2,3,6,7,8,13,16,17,18,21,22,23 channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.3.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.3.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.4.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.4.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.6.conv.2,features.5.conv.2,features.4.conv.2
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 0,3,7,9,10,11,15,16,18,20,21,22,25,27,28,29 channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.5.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.5.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.6.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.6.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.7.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.7.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.7.conv.2,features.9.conv.2,features.10.conv.2,features.8.conv.2
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 0,1,2,4,5,8,9,10,12,14,16,25,28,29,30,32,33,38,39,41,42,43,45,47,48,49,50,53,55,60,61,63 channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.8.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.8.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.9.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.9.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.10.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.10.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.11.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.11.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.13.conv.2,features.11.conv.2,features.12.conv.2
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 0,1,2,4,6,7,10,11,13,15,18,19,20,21,22,23,24,28,29,30,34,36,38,40,43,46,47,48,52,56,57,58,62,63,66,69,70,72,74,79,81,82,87,88,89,90,92,95 channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.12.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.12.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.13.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.13.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.14.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.14.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.15.conv.2,features.14.conv.2,features.16.conv.2
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 0,1,6,7,9,12,14,18,20,22,25,26,28,32,33,34,35,37,39,40,41,46,48,50,51,58,62,64,65,66,67,69,70,71,72,73,75,76,77,78,79,80,82,87,88,89,95,96,101,102,103,111,114,119,120,121,122,123,125,126,127,128,130,131,132,134,135,136,137,139,140,142,143,145,147,148,150,152,156,158 channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.15.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.15.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.16.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.16.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.17.conv.0.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.17.conv.1.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the  channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.17.conv.2
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 1,2,3,5,8,10,12,15,16,17,21,22,24,25,27,29,31,32,34,36,38,39,40,43,44,45,48,49,51,58,59,61,62,64,66,67,72,73,74,75,77,78,80,82,84,85,86,88,89,92,93,95,97,100,103,105,107,108,115,116,117,119,121,123,125,127,128,129,130,132,134,135,136,138,139,140,141,143,145,151,152,154,155,156,157,158,162,164,165,166,167,168,170,171,172,173,182,183,184,185,186,188,190,191,192,193,195,196,198,200,204,205,207,208,211,212,214,215,219,222,226,227,229,231,232,239,240,242,243,245,253,254,257,260,261,263,264,266,269,276,277,279,282,284,285,286,287,289,292,295,297,298,300,302,307,308,314,317,318,319 channels for all dependent
[2021-04-07 14:12:08] INFO (nni.algorithms.compression.pytorch.pruning.one_shot/MainThread) Pruning the dependent layers: features.18.0
[2021-04-07 14:12:08] INFO (torch filter pruners/MainThread) Prune the 2,5,7,8,10,11,12,13,14,17,18,20,24,25,26,28,29,30,31,33,34,35,37,38,40,42,43,44,46,47,48,49,50,54,56,60,61,62,71,72,73,74,75,80,82,83,85,89,91,93,94,97,98,103,104,106,107,109,113,114,117,119,120,121,122,123,126,127,128,130,131,133,135,136,138,140,142,147,148,149,151,153,154,155,156,160,161,167,168,169,171,172,173,174,178,179,180,181,182,183,185,186,187,188,189,190,191,192,193,194,196,198,199,200,201,203,204,207,210,212,214,220,225,226,227,228,229,230,232,233,234,235,236,239,240,242,243,244,250,253,256,257,258,259,260,261,262,264,265,268,269,270,274,284,285,289,292,296,300,301,304,305,306,309,310,311,315,316,317,319,322,327,328,329,331,332,335,336,337,339,340,345,346,347,351,354,358,361,363,365,369,371,374,376,379,380,381,383,384,388,389,391,393,394,396,397,400,401,406,407,410,416,417,419,420,422,424,426,427,428,429,430,435,436,437,438,439,441,444,446,447,449,452,455,456,459,462,466,469,470,476,479,480,481,483,484,486,487,488,489,495,496,497,498,500,503,505,506,507,510,511,512,513,516,519,521,522,523,525,532,534,537,538,539,541,542,543,547,548,551,554,555,556,560,563,565,566,568,569,570,571,572,573,574,575,576,578,580,581,585,586,587,588,590,591,595,596,597,601,603,604,605,606,608,609,610,612,616,618,619,621,625,626,629,630,633,636,637,638,642,643,646,649,650,652,653,655,657,661,662,663,664,665,666,669,671,673,674,676,677,678,679,680,685,687,688,690,692,693,694,695,697,701,702,709,710,711,714,715,717,719,720,722,723,725,726,727,733,735,740,741,742,743,746,747,748,751,753,755,756,757,762,763,764,765,766,767,770,772,774,776,779,780,782,783,785,786,787,792,793,794,795,799,801,802,804,805,807,810,814,816,819,820,821,822,823,826,832,833,834,835,836,843,846,847,853,854,855,859,863,865,866,867,868,870,875,881,883,884,887,889,890,891,894,899,900,903,906,907,908,909,912,913,917,918,919,922,923,925,926,929,930,931,932,934,935,937,939,945,947,949,950,953,954,956,959,961,963,964,965,966,967,971,975,976,977,981,983,985,988,990,992,993,995,998,1000,1004,1006,1008,1009,1010,1011,1013,1016,1017,1018,1020,1021,1022,1026,1030,1034,1039,1041,1042,1043,1044,1045,1046,1048,1049,1050,1051,1053,1058,1061,1064,1065,1067,1069,1070,1071,1072,1073,1076,1077,1083,1084,1085,1086,1088,1092,1099,1100,1101,1102,1104,1109,1111,1113,1114,1116,1117,1118,1119,1120,1121,1125,1128,1129,1130,1132,1134,1139,1140,1141,1142,1143,1152,1153,1155,1158,1159,1165,1166,1167,1168,1173,1175,1176,1177,1180,1183,1188,1190,1191,1192,1196,1197,1198,1200,1202,1204,1206,1207,1209,1212,1214,1219,1220,1221,1223,1224,1226,1227,1229,1232,1233,1234,1239,1242,1244,1245,1246,1253,1256,1257,1258,1259,1260,1261,1262,1264,1265,1266,1267,1274,1276,1277,1278 channels for all dependent

===========log over ============
